src/kernel/cuda/binary_reduce_impl.cuh

```diff
@@ -72,6 +72,53 @@ __device__ __forceinline__ int64_t Ravel(
   return out;
 }
 
+/*
+ * Merge convert flattened index to multi-dimension index according to output shape
+ * and Convert multi-dimension index to flattened index for lhs or rhs (assume row-major).
+ */
+__device__ __forceinline__ void URRavel(
+    int64_t idx, int ndim, const int64_t* shape, const int64_t* stride,
+    const int64_t* lhs_shape, const int64_t* lhs_stride,
+    const int64_t* rhs_shape, const int64_t* rhs_stride, int64_t &lhs_out, int64_t &rhs_out) {
+  if (stride[0] == lhs_stride[0]) {
+#pragma unroll
+    for (int d = 0; d < ndim; ++d) {
+      int64_t o_sh = shape[d];
+      int64_t o_st = stride[d];
+      int64_t rhs_sh = rhs_shape[d];
+      int64_t rhs_st = rhs_stride[d];
+      
+      int64_t i = (idx / o_st) % o_sh;
+      /*
+       * Simplfied for rhs_out += min(i, rhs_sh - 1) * rhs_st;
+       * rhs_sh be N or 1
+       */
+      if (rhs_sh > i) { 
```

Comment:
I'm not sure which one is more intuitive, if you use `if` style, I suggest not to mention `min(i, rhs_sh - 1) * rhs_st` in the doc, which might be confusing.

Reply:
I think rhs_out += min(i, rhs_sh - 1) * rhs_st; is more intuitive to understand.
So I left it in the comment.